src/index.ts

```diff
@@ -1,36 +1,52 @@
-const { promisify } = require('promise-callbacks');
+import { promisify } from 'promise-callbacks';
 
-async function promiseIterate(iterable, iteratee, { batchSize } = {}) {
+type Iterable = {
+  next: (p?: any) => Promise<any | void>;
+};
+type Iteratee = (p: any) => Promise<any>;
```

Comment:
Does it make sense to use generics here? I'm not fully sure on functionality, but something like:

```suggestion
type Iterable<T> = {
  next: (p?: T) => Promise<T | void>;
};
type Iteratee<T> = (p: T) => Promise<T>;
```

Reply:
absolutely! amended on my last commit, tks!